src/huggingface_hub/eval_results.py

```diff
@@ -0,0 +1,211 @@
+        verify_token (`str`, *optional*):
+            Cryptographic proof that the evaluation is auditable and reproducible.
+        date (`str`, *optional*):
+            When the evaluation was run (ISO-8601 datetime). Defaults to git commit time.
+        source_url (`str`, *optional*):
+            Link to the evaluation source (e.g., logs). Required if `source_name` or `source_user` is provided.
+        source_name (`str`, *optional*):
+            Display name for the source. Example: "Eval Logs".
+        source_user (`str`, *optional*):
+            HF user or org name for attribution. Example: "cais".
+
+    Example:
+        ```python
+        >>> from huggingface_hub import EvalResultEntry
+        >>> # Minimal example with required fields only
+        >>> result = EvalResultEntry(
+        ...     dataset_id="Idavidrein/gpqa",
+        ...     value=0.412,
+        ...     task_id="gpqa_diamond",
+        ... )
+        >>> # Full example with all fields
+        >>> result = EvalResultEntry(
+        ...     dataset_id="cais/hle",
+        ...     value=20.90,
+        ...     task_id="default",
+        ...     dataset_revision="5503434ddd753f426f4b38109466949a1217c2bb",
+        ...     verify_token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        ...     date="2025-01-15T10:30:00Z",
+        ...     source_url="https://huggingface.co/datasets/cais/hle",
+        ...     source_name="CAIS HLE",
+        ...     source_user="cais",
+        ... )
+
+        ```
+    """
+
+    dataset_id: str
+    value: Any
+    task_id: Optional[str] = None
+    dataset_revision: Optional[str] = None
+    verify_token: Optional[str] = None
+    date: Optional[str] = None
+    source_url: Optional[str] = None
+    source_name: Optional[str] = None
+    source_user: Optional[str] = None
+
+    def __post_init__(self) -> None:
+        if (self.source_name is not None or self.source_user is not None) and self.source_url is None:
+            raise ValueError("If `source_name` or `source_user` is provided, `source_url` must also be provided.")
+
+
+def eval_result_entries_to_yaml(entries: list[EvalResultEntry]) -> list[dict[str, Any]]:
+    """Convert a list of [`EvalResultEntry`] objects to a YAML-serializable list of dicts.
+
+    This produces the format expected in `.eval_results/*.yaml` files.
+
+    Args:
+        entries (`list[EvalResultEntry]`):
+            List of evaluation result entries to serialize.
+
+    Returns:
+        `list[dict[str, Any]]`: A list of dictionaries ready to be dumped to YAML.
+
+    Example:
+        ```python
+        >>> from huggingface_hub import EvalResultEntry, eval_result_entries_to_yaml
+        >>> entries = [
+        ...     EvalResultEntry(dataset_id="cais/hle", value=20.90, task_id="default"),
+        ...     EvalResultEntry(dataset_id="Idavidrein/gpqa", value=0.412, task_id="gpqa_diamond"),
+        ... ]
+        >>> yaml_data = eval_result_entries_to_yaml(entries)
+        >>> yaml_data[0]
```

Comment:
maybe complete the example to save that to a `.eval_results/...` file on the repo on the Hub. I don't think it makes sense to provide a helper dedicated to that but highlighting how to use it would be great